### PR TITLE
Fixing bug of ekobox evolve_pdfs with multiple replicas

### PR DESCRIPTION
**Pull request by andreab1997**
Solve #130 
The bug was due to the fact that, even if `all_blocks` was supposed to contain all and only the blocks of a certain replica, it was not initialized to zero at the beginning of each loop step on `evolved_PDFs`.

### CODE REVIEW

**Comment by felixhekhorn** [comment]
Closes #130 

Github has some [keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) to automatically close an associated issue

**Comment by andreab1997** [comment]
I believe that now the problem should be solved. Thanks to @giacomomagni to have spotted this. :)

File: benchmarks/ekobox/benchmark_evol_pdf.py [hunk 1]
```diff
@@ -91,8 +91,14 @@ def benchmark_evolve_more_members(tmp_path, cd, lhapdf_path):
             ev_p.evolve_pdfs(pdfs, theory, op, install=True, name="Debug")
         # ev_pdfs
         _ = lhapdf.mkPDFs("Debug")
```
Comment (AleCandido) [inline benchmarks/ekobox/benchmark_evol_pdf.py:93]: Maybe now you don't need any longer this line

Comment (AleCandido) [inline benchmarks/ekobox/benchmark_evol_pdf.py:93]: Maybe now you don't need any longer this line

Comment (andreab1997) [reply benchmarks/ekobox/benchmark_evol_pdf.py:93]: I would keep it because it is testing if lhapdf is able to load the complete set

Comment (AleCandido) [reply benchmarks/ekobox/benchmark_evol_pdf.py:93]: Is it doing something different than loading the info and the individual replicas?

Comment (AleCandido) [reply benchmarks/ekobox/benchmark_evol_pdf.py:93]: If so, don't discard the return value, but test what you expect it to be.

Comment (andreab1997) [reply benchmarks/ekobox/benchmark_evol_pdf.py:93]: No it is only loading the info and all the replicas and constructing the lhapdf object. Maybe I can just check if the number of final replicas is the same of the initial number

**Comment by giacomomagni** [comment]
Okay thanks for fixing it! Can I add also the feature to store the computed eko ?
Right now evolve_pdfs loads the eko but has not the option to store it...

File: benchmarks/ekobox/benchmark_evol_pdf.py (hunk 1 above)
Comment (andreab1997) [reply benchmarks/ekobox/benchmark_evol_pdf.py]: Have a look if now it is ok

**Review by AleCandido — APPROVED** [review]

**Review by felixhekhorn — APPROVED** [review]

**Review by felixhekhorn — APPROVED** [review]

**Pull request merged**